### PR TITLE
feat(input, input-text, input-number): add attributes autocomplete, accept, multiple, pattern

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #4079 

## Summary
Adds native input attributes:

- [multiple](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/multiple) to `calcite-input` for the types specified on MDN
- [accept](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/accept)  to `calcite-input` for the types specified on MDN
- [pattern](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/pattern) to `calcite-input-text` and  `calcite-input` for the types specified on MDN
- [autocomplete](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete) to `calcite-input-number`, `calcite-input-text`, and  `calcite-input` for the types specified on MDN

## Note
 just saw on MDN after committing that Safari may not support `autocomplete`:
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete#browser_compatibility

Should I remove it from the PR?

### CODE REVIEW

File: src/components/input-number/input-number.tsx [hunk 1]
```diff
@@ -258,6 +258,14 @@ export class InputNumber
    */
   @Prop({ reflect: true }) step?: number | "any";
 
+  /**
+   * Specifies the type of content to autocomplete, for use in forms.
+   * Read the native attribute's documentation on MDN for more info.
+   *
+   * @mdn [step](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete)
+   */
+  @Prop({ reflect: true }) autocomplete?: "string";
```
Comment (driskull) [inline src/components/input-number/input-number.tsx:267]: @benelan shouldn't it be `autocomplete: string;`?

We don't need the optional question mark and the type should be string not "string" for these right?

Comment (driskull) [reply src/components/input-number/input-number.tsx:267]: Do these need to be reflected?

Comment (benelan) [reply src/components/input-number/input-number.tsx:267]: Woops my bad, fixing. I copy and pasted from the `step` property which has "any" in quotes for some reason. I'll fix that in a separate PR

**Review by driskull — APPROVED** [review]
Looks good!

Do you think it would beneficial to make sure these are passed to their internal inputs via tests? if not i'm ok with that.

**Comment by benelan** [comment]
Good idea, I'll add some coverage. I did test them in the demos

**Pull request merged**